utils/get_supported_scylla_base_versions.py

```diff
@@ -0,0 +1,137 @@
+start_support_versions = {'centos-8': {'scylla': '4.1', 'enterprise': '2021.1'}}
+
+
+class UpgradeBaseVersion:  # pylint: disable=too-many-instance-attributes
+
+    oos_start_support_version = None
+    ent_start_support_version = None
+
+    def __init__(self, scylla_repo: str, linux_distro: str, scylla_version: str = None):
+        if len(linux_distro.split('-')) > 1:
+            self.dist_type, *_, self.dist_version = linux_distro.split('-')
+        else:
+            self.dist_type = linux_distro.split('-')[0]
+            self.dist_version = None
+        self.scylla_repo = scylla_repo
+        self.linux_distro = linux_distro
+        self.product, self.scylla_version = self.get_product_and_version(scylla_version)
+        self.repo_maps = get_s3_scylla_repos_mapping(self.dist_type, self.dist_version)
+
+    def get_product_and_version(self, scylla_version: str = None):
+        """
+        return scylla product name and major version. if scylla_version isn't assigned,
+        we will try to get the major version from the scylla_repo url.
+        """
+        if scylla_version is None:
+            assert 'unstable/' in self.scylla_repo
+            product, version = self.scylla_repo.split('unstable/')[1].split('/')[0:2]
+            scylla_version = version.replace('branch-', '').replace('enterprise-', '')
+        else:
+            product = 'scylla-enterprise' if is_enterprise(scylla_version) else 'scylla'
+        return product, scylla_version
+
+    def get_start_support_version(self):
+        """
+        We can't detect the support versions for this distro, which shares the repo with others, eg: centos8
+        so we need to assign the start support versions for it.
+        """
+        if (self.dist_type not in ['debian', 'ubuntu'] and self.linux_distro != 'centos'):
+            versions_dict = start_support_versions.get(self.linux_distro, None)
+            if versions_dict is None:
+                raise Exception(
+                    f"We can't detect the support versions for this distro, which shares the repo with {self.dist_type}.")
+            self.oos_start_support_version = versions_dict['scylla']
+            self.ent_start_support_version = versions_dict['enterprise']
+
+    def get_supported_scylla_base_versions(self, supported_versions):  # pylint: disable=too-many-branches
+        """
+        We have special base versions list for each release, and we don't support to upgraded from enterprise
+        to opensource. This function is used to get the base versions list which will be used in the upgrade test.
+
+        @supported_version: all scylla release versions, the base versions will be filtered out from the supported_version
+        """
+        oos_base_version = []
+        ent_base_version = []
+
+        oos_release_list = [i for i in supported_versions if not is_enterprise(i)]
+        ent_release_list = [i for i in supported_versions if is_enterprise(i)]
```

Comment:
Small nit: "i" is used as step item for "Integer"s. "s" for "String"s. But also, instead of string we may use other. In currect case it may be "version" or just "v".

Reply:
Good suggestion, fixed.